pkg/repository/udmrepo/kopialib/backend/utils.go

```diff
@@ -0,0 +1,77 @@
+/*
+Copyright the Velero contributors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package backend
+
+import (
+	"strconv"
+	"time"
+
+	"github.com/pkg/errors"
+)
+
+func mustHaveString(key string, flags map[string]string) (string, error) {
+	if value, exist := flags[key]; exist {
+		return value, nil
+	} else {
+		return "", errors.New("key " + key + " not found")
+	}
+}
+
+func optionalHaveString(key string, flags map[string]string) string {
+	return optionalHaveStringWithDefault(key, flags, "")
+}
+
+func optionalHaveBool(key string, flags map[string]string) bool {
+	if value, exist := flags[key]; exist {
+		ret, err := strconv.ParseBool(value)
```

Comment:
Although it's optional we should not eat the error, same issue for other `optionalHavexxx` funcs in this file.

Reply:
Done, log the conversion error in place